x-pack/plugins/security_solution/server/endpoint/routes/resolver/utils/children_helper.test.ts

```diff
@@ -3,78 +3,195 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-import _ from 'lodash';
-
-import { EndpointDocGenerator } from '../../../../../common/endpoint/generate_data';
+import {
+  EndpointDocGenerator,
+  Tree,
+  Event,
+  TreeNode,
+} from '../../../../../common/endpoint/generate_data';
 import { ChildrenNodesHelper } from './children_helper';
-import { eventId, entityId, parentEntityId } from '../../../../../common/endpoint/models/event';
-import { ResolverEvent, ResolverChildren } from '../../../../../common/endpoint/types';
-
-function findParents(events: ResolverEvent[]): ResolverEvent[] {
-  const cache = _.groupBy(events, entityId);
+import { eventId, isProcessStart } from '../../../../../common/endpoint/models/event';
 
-  const parents: ResolverEvent[] = [];
-  Object.values(cache).forEach((lifecycle) => {
-    const parentNode = cache[parentEntityId(lifecycle[0])!];
-    if (parentNode) {
-      parents.push(parentNode[0]);
+function getStartEvents(events: Event[]): Event[] {
+  const startEvents: Event[] = [];
+  for (const event of events) {
+    if (isProcessStart(event)) {
+      startEvents.push(event);
     }
-  });
-  return parents;
+  }
+  return startEvents;
 }
 
-function findNode(tree: ResolverChildren, id: string) {
-  return tree.childNodes.find((node) => {
-    return node.entityID === id;
-  });
+function getAllChildrenEvents(tree: Tree) {
+  const children: Event[] = [];
+  for (const child of tree.children.values()) {
+    children.push(...child.lifecycle);
+  }
+  return children;
+}
+
+function getStartEventsFromLevels(levels: Array<Map<string, TreeNode>>) {
```

Comment:
Wonder what this performance is gonna look like at scale 😅.

Reply:
This just a test file :)

Reply:
that'll do it 😂